src/Symfony/Component/Amqp/Helper/MessageExporter.php

```diff
@@ -0,0 +1,88 @@
+<?php
+
+/*
+ * This file is part of the Symfony package.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Symfony\Component\Amqp\Helper;
+
+use Symfony\Component\Amqp\Broker;
+use Symfony\Component\Amqp\Exception\InvalidArgumentException;
+
+/**
+ * An utility class to return a compressed file with all
+ * message for a Queue.
+ *
+ * @author Grégoire Pineau <lyrixx@lyrixx.info>
+ */
+class MessageExporter
```

Comment:
I'm not a heavy user of queues, so I can be totally wrong ... but this command looks strange to me. Why not allowing to export all queue messages as JSON files in some dir. If the user wants compression too, they can use any command tool. Compressing messages (and picking the tgz format specifically) looks a bit "too much" for Symfony core.

Reply:
AMQP Messages could be something else than JSON, It could be an image for exemple
So exporting that in a json file is too opinionated.

But indeed, the compression could be an option, but it comes for almost free ;)

But you are right about ` "too much" for Symfony core.` It was from our internal sensiolabs/amqp package. but it's really useful to debug ;)